tests.foreman.cli.test_ostreebranch.OstreeBranchTestCase.test_positive_list_by_repo_id

Reason (thanks @rplevka ):

#5305

Stack Trace:

Error Message
AssertionError: 0 not greater than 0
Stacktrace
args = (<tests.foreman.cli.test_ostreebranch.OstreeBranchTestCase testMethod=test_positive_list_by_repo_id>,)
kwargs = {}, settings_project = 'sat'

    @wraps(func)
    def wrapper(*args, **kwargs):
        """Wrapper that will skip the test if the test method project does
                not match with the settings project.
                """
        # Validate project value
        if settings.project:
            settings_project = settings.project.lower()
        else:
            settings_project = 'sat'
    
        if project not in allowed_project_modes:
            raise ProjectModeError(
                '"{0}" is not a project mode. The allowed project modes '
                'are: {1}'.format(project, allowed_project_modes)
            )
    
        # If robottelo.properties not present or does not specify a project
        # use sat
        if settings_project not in allowed_project_modes:
            raise ProjectModeError(
                '"{0}" is not an acceptable "[robottelo] project" value '
                'in robottelo.properties file. The allowed project modes '
                'are: {1}'.format(settings_project, allowed_project_modes)
            )
    
        # Preconditions PASS.  Now skip the test if modes does not match
        if project != settings_project:
            raise unittest2.SkipTest(
                'Server runs in "{0}" mode and this test will run '
                'only on "{1}" mode.'.format(settings_project, project)
            )
        else:
>           return func(*args, **kwargs)

robottelo/decorators/__init__.py:268: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/foreman/cli/test_ostreebranch.py:88: in test_positive_list_by_repo_id
    self.assertGreater(len(result), 0)
E   AssertionError: 0 not greater than 0

--- a/robottelo/cli/factory.py
+++ b/robottelo/cli/factory.py
@@ -148,6 +148,19 @@ def create_object(cli_object, options, values):
     return result
 
 
+def _entity_with_credentials(credentials, cli_entity_cls):
+    """Create entity class using credentials. If credentials is None will
+    return cli_entity_cls itself
+
+    :param credentials: tuple (login, password)
+    :param cli_entity_cls: Cli Entity Class
+    :return: Cli Entity Class
+    """
+    if credentials is not None:
+        cli_entity_cls = cli_entity_cls.with_user(*credentials)
+    return cli_entity_cls
+
+
 @cacheable
 def make_activation_key(options=None):
     """
@@ -318,6 +331,36 @@ def make_content_view(options=None):
         --name NAME                   Name of the content view
         --organization ORGANIZATION_NAME  Organization name to search by
         --organization-id ORGANIZATION_ID Organization identifier
+        --organization-label ORGANIZATION_LABEL Organization label to
+        search by
+        --product PRODUCT_NAME          Product name to search by
+        --product-id PRODUCT_ID         product numeric identifier
+        --repositories REPOSITORY_NAMES Comma separated list of values.
+        --repository-ids REPOSITORY_IDS List of repository ids
+                                        Comma separated list of values.
+        -h, --help                    print help
+
+    """
+    return make_content_view_with_credentials(options)
+
+
+def make_content_view_with_credentials(options=None, credentials=None):
+    """
+    Usage::
+
+        hammer content-view create [OPTIONS]
+
+    Options::
+
+        --component-ids COMPONENT_IDS List of component content view
+        version ids for composite views
+                                      Comma separated list of values.
+        --composite                   Create a composite content view
+        --description DESCRIPTION     Description for the content view
+        --label LABEL                 Content view label
+        --name NAME                   Name of the content view
+        --organization ORGANIZATION_NAME  Organization name to search by
+        --organization-id ORGANIZATION_ID Organization identifier
         --organization-label ORGANIZATION_LABEL Organization label to search by
         --product PRODUCT_NAME          Product name to search by
         --product-id PRODUCT_ID         product numeric identifier
@@ -325,6 +368,9 @@ def make_content_view(options=None):
         --repository-ids REPOSITORY_IDS List of repository ids
                                         Comma separated list of values.
         -h, --help                    print help
+
+    If credentials is None default credentials present on
+    robottelo.properties will be used.
 
     """
     # Organization ID is a required field.
@@ -346,7 +392,8 @@ def make_content_view(options=None):
         u'repository-ids': None
     }
 
-    return create_object(ContentView, args, options)
+    cv_cls = _entity_with_credentials(credentials, ContentView)
+    return create_object(cv_cls, args, options)
 
 
 @cacheable
@@ -753,6 +800,10 @@ def make_partition_table(options=None):
 
 @cacheable
 def make_product(options=None):
+    return make_product_with_credentials(options)
+
+
+def make_product_with_credentials(options=None, credentials=None):
     """
     Usage::
 
@@ -788,8 +839,8 @@ def make_product(options=None):
         u'sync-plan': None,
         u'sync-plan-id': None,
     }
-
-    return create_object(Product, args, options)
+    product_cls = _entity_with_credentials(credentials, Product)
+    return create_object(product_cls, args, options)
 
 
 def make_product_wait(options=None, wait_for=5):
@@ -892,6 +943,10 @@ def make_registry(options=None):
 
 @cacheable
 def make_repository(options=None):
+    return make_repository_with_credentials(options)
+
+
+def make_repository_with_credentials(options=None, credentials=None):
     """
     Usage::
 
@@ -966,8 +1021,8 @@ def make_repository(options=None):
         u'publish-via-http': u'true',
         u'url': FAKE_1_YUM_REPO,
     }
-
-    return create_object(Repository, args, options)
+    repo_cls = _entity_with_credentials(credentials, Repository)
+    return create_object(repo_cls, args, options)
 
 
 @cacheable
@@ -2040,6 +2095,10 @@ def make_compute_resource(options=None):
 
 @cacheable
 def make_org(options=None):
+    return make_org_with_credentials(options)
+
+
+def make_org_with_credentials(options=None, credentials=None):
     """
     Usage::
 
@@ -2135,8 +2194,8 @@ def make_org(options=None):
         u'user-ids': None,
         u'users': None,
     }
-
-    return create_object(Org, args, options)
+    org_cls = _entity_with_credentials(credentials, Org)
+    return create_object(org_cls, args, options)
 
 
 @cacheable

--- a/tests/foreman/cli/test_ostreebranch.py
+++ b/tests/foreman/cli/test_ostreebranch.py
@@ -17,12 +17,14 @@
 """
 
 import random
+
 from robottelo.cli.contentview import ContentView
 from robottelo.cli.factory import (
     make_content_view,
-    make_org,
-    make_product,
-    make_repository,
+    make_org_with_credentials,
+    make_product_with_credentials,
+    make_repository_with_credentials,
+    make_user
 )
 from robottelo.cli.ostreebranch import OstreeBranch
 from robottelo.cli.repository import Repository
@@ -41,22 +43,44 @@ class OstreeBranchTestCase(CLITestCase):
     def setUpClass(cls):
         """Create an organization, product and ostree repo."""
         super(OstreeBranchTestCase, cls).setUpClass()
-        cls.org = make_org()
-        cls.product = make_product({u'organization-id': cls.org['id']})
+        password = 'password'
+        cls.user = make_user({'admin': 'true', 'password': password})
+        cls.user['password'] = password
+        credentials = cls.get_user_credentials()
+        cls.org = make_org_with_credentials(
+            credentials=credentials)
+        cls.product = make_product_with_credentials(
+            {
+                u'organization-id': cls.org['id']
+            },
+            credentials
+        )
         # Create new custom ostree repo
-        cls.ostree_repo = make_repository({
-            u'product-id': cls.product['id'],
-            u'content-type': u'ostree',
-            u'publish-via-http': u'false',
-            u'url': FEDORA23_OSTREE_REPO,
-        })
-        Repository.synchronize({'id': cls.ostree_repo['id']})
-        cls.cv = make_content_view({
-            u'organization-id': cls.org['id'],
-            u'repository-ids': [cls.ostree_repo['id']],
-        })
-        ContentView.publish({u'id': cls.cv['id']})
-        cls.cv = ContentView.info({u'id': cls.cv['id']})
+        cls.ostree_repo = make_repository_with_credentials(
+            {
+                u'product-id': cls.product['id'],
+                u'content-type': u'ostree',
+                u'publish-via-http': u'false',
+                u'url': FEDORA23_OSTREE_REPO,
+            },
+            credentials
+        )
+        Repository.with_user(*credentials).synchronize(
+            {'id': cls.ostree_repo['id']})
+        cls.cv = make_content_view(
+            {
+                u'organization-id': cls.org['id'],
+                u'repository-ids': [cls.ostree_repo['id']],
+            },
+            credentials
+        )
+        ContentView.with_user(*credentials).publish({u'id': cls.cv['id']})
+        cls.cv = ContentView.with_user(*credentials).info(
+            {u'id': cls.cv['id']})
+
+    @classmethod
+    def get_user_credentials(cls):
+        return cls.user['login'], cls.user['password']
 
     @run_only_on('sat')
     @tier1
@@ -69,7 +93,7 @@ class OstreeBranchTestCase(CLITestCase):
 
         :CaseImportance: Critical
         """
-        result = OstreeBranch.list()
+        result = OstreeBranch.with_user(*self.get_user_credentials()).list()
         self.assertGreater(len(result), 0)
 
     @run_only_on('sat')
@@ -84,7 +108,9 @@ class OstreeBranchTestCase(CLITestCase):
 
         :CaseImportance: Critical
         """
-        result = OstreeBranch.list({'repository-id': self.ostree_repo['id']})
+
+        branch = OstreeBranch.with_user(*self.get_user_credentials())
+        result = branch.list({'repository-id': self.ostree_repo['id']})
         self.assertGreater(len(result), 0)
 
     @run_only_on('sat')
@@ -98,7 +124,8 @@ class OstreeBranchTestCase(CLITestCase):
 
         :CaseImportance: Critical
         """
-        result = OstreeBranch.list({'product-id': self.product['id']})
+        result = OstreeBranch.with_user(*self.get_user_credentials()).list(
+            {'product-id': self.product['id']})
         self.assertGreater(len(result), 0)
 
     @run_only_on('sat')
@@ -112,7 +139,8 @@ class OstreeBranchTestCase(CLITestCase):
 
         :CaseImportance: Critical
         """
-        result = OstreeBranch.list({'organization-id': self.org['id']})
+        result = OstreeBranch.with_user(*self.get_user_credentials()).list(
+            {'organization-id': self.org['id']})
         self.assertGreater(len(result), 0)
 
     @run_only_on('sat')
@@ -126,7 +154,8 @@ class OstreeBranchTestCase(CLITestCase):
 
         :CaseImportance: Critical
         """
-        result = OstreeBranch.list({'content-view-id': self.cv['id']})
+        result = OstreeBranch.with_user(*self.get_user_credentials()).list(
+            {'content-view-id': self.cv['id']})
         self.assertGreater(len(result), 0)
 
     @run_only_on('sat')
@@ -140,9 +169,10 @@ class OstreeBranchTestCase(CLITestCase):
 
         :CaseImportance: Critical
         """
-        result = OstreeBranch.list()
+        result = OstreeBranch.with_user(*self.get_user_credentials()).list()
         self.assertGreater(len(result), 0)
         # Grab a random branch
         branch = random.choice(result)
-        result = OstreeBranch.info({'id': branch['id']})
+        result = OstreeBranch.with_user(*self.get_user_credentials()).info(
+            {'id': branch['id']})
         self.assertEqual(branch['id'], result['id'])

